builtin/logical/aws/rotation.go

```diff
@@ -0,0 +1,125 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/vault/sdk/logical"
+	"github.com/hashicorp/vault/sdk/queue"
+)
+
+func (b *backend) initQueue(_ context.Context, _ *logical.InitializationRequest) error {
+	return nil
+}
+
+func (b *backend) rotateExpiredStaticCreds(ctx context.Context, req *logical.Request) error {
+	var errs *multierror.Error
+
+	for {
+		keepGoing, err := b.rotateCredential(ctx, req.Storage)
+		if err != nil {
+			errs = multierror.Append(errs, err)
+		}
+		if !keepGoing {
+			if errs != nil && len(errs.Errors) != 0 {
+				return fmt.Errorf("error(s) occurred while rotating expired static credentials: %w", errs)
+			} else {
+				return nil
+			}
+		}
+	}
+}
+
+func (b *backend) rotateCredential(ctx context.Context, storage logical.Storage) (bool, error) {
```

Comment:
without reading how this method is used, it's a bit hard to determine what the `bool` return value represents. I wonder if this justifies using named return parameters, perhaps `(rotated bool, err error)`?